File: frontend/types/menu.ts
// 메뉴 이미지
export interface MenuImage {
  id: string;
  url: string;
  isPrimary: boolean;
  sortOrder: number;
  file?: File; // 프론트엔드 업로드용
}

// 옵션 아이템
export interface OptionItem {
  id: string;
  name: string;
  priceDelta: number; // 추가 가격 (0이면 무료)
}

// 메뉴 옵션
export interface MenuOption {
  id: string;
  name: string;
  type: 'radio' | 'checkbox';
  required: boolean;
  items: OptionItem[];
}

// 메뉴 카테고리
export interface MenuCategory {
  id: string;
  korName: string;
  engName: string;
  icon?: string;
  sortOrder: number;
}

// 메뉴
export interface Menu {
  id: string;
  korName: string;
  engName: string;
  description: string;
  price: number;
  category: MenuCategory;
  images: MenuImage[];
  isAvailable: boolean;
  isSoldOut: boolean;
  sortOrder: number;
  options: MenuOption[];
  createdAt: Date;
  updatedAt: Date;
}

// 메뉴 생성/수정용 DTO
export interface MenuFormData {
  korName: string;
  engName: string;
  description: string;
  price: number;
  categoryId: string;
  images: MenuImage[];
  isAvailable: boolean;
  isSoldOut: boolean;
  options: MenuOption[];
}

File: backend/src/main/java/com/new_cafe/app/backend/menu/domain/model/Menu.java
package com.new_cafe.app.backend.menu.domain.model;

import java.time.LocalDateTime;
import java.util.List;

import com.new_cafe.app.backend.category.domain.model.Category;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

/**
 * 메뉴 도메인 모델 (순수 Java - 프레임워크 의존성 없음)
 */
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Menu {
    private Long id;
    private String korName;
    private String engName;
    private String description;
    private Integer price;
    private Long categoryId;
    private Boolean isAvailable;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // 연관 도메인 (필요 시 로딩)
    private Category category;

    private List<MenuImage> images;
}
